cli/default-reporter/test/tsconfig.json

```diff
@@ -0,0 +1,38 @@
+{
+  "extends": "../tsconfig.json",
+  "compilerOptions": {
+    "composite": false,
+    "noEmit": true,
+    "rootDir": "."
+  },
+  "include": [
+    "**/*.ts",
+    "../../../__typings__/**/*.d.ts"
+  ],
+  "references": [
+    {
+      "path": ".."
+    },
+    {
+      "path": "../../../config/config"
+    },
+    {
+      "path": "../../../dedupe/issues-renderer"
+    },
+    {
+      "path": "../../../dedupe/types"
+    },
+    {
+      "path": "../../../packages/core-loggers"
+    },
+    {
+      "path": "../../../packages/error"
+    },
+    {
+      "path": "../../../packages/render-peer-issues"
+    },
+    {
+      "path": "../../../packages/types"
+    }
```

Comment:
Presumably this is just the same list as the parent project; maybe these can be omitted?

Reply:
I don't think there's a way to inherit `extends` the parent `references` array?

https://www.typescriptlang.org/tsconfig/#extends

> Currently, the only top-level property that is excluded from inheritance is [references](https://www.typescriptlang.org/tsconfig/#references).

This list is similar, but also has `{ "path": ".." }` to add a project reference to the main implementation. Here's where this list is generated after the changes in this PR:

https://github.com/pnpm/pnpm/blob/885cae28176d656af3b2db4e5127355498681f93/.meta-updater/src/index.ts#L146-L161

Reply:
Right, but you're depending on a project that depends on the other projects. Unless this script is intentionally including all transitive deps for all packages? I don't think that it is?

Reply:
Oh right. 🤔

The tests can directly import other workspace packages.

https://github.com/pnpm/pnpm/blob/a2f5f5c990003d2b456a1e6920258f746ff49e5a/exec/lifecycle/test/index.ts#L4

So I thought the project references needed to be at the same level for build-free editing to work. But testing this out, just the `{ path: ".." }` reference is enough to make these transitives work as expected.

Great suggestion! 🙇🏻‍♂️

Going to push an update that makes this PR much smaller.

Reply:
It's not all perfect, it's possible that it won't work due to https://github.com/microsoft/TypeScript/issues/30608.

Reply:
The description in the issue linked is a bit surprising. I would expect `tsc --build` to go through transitives and build their `.d.ts` files. I feel like a test repo would help me understand that issue, but that wasn't included.

In any case, your suggestion does seem to work here and is much cleaner. 🙂